File: golang/uPIMulator/src/linker/kernel/instruction/cc/acquire_cc.go
package cc

import (
	"errors"
)

type AcquireCc struct {
	condition Condition
}

func (this *AcquireCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		Z:    true,
		NZ:   true,
		TRUE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *AcquireCc) Condition() Condition {
	return this.condition
}

File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/acquire_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class AcquireCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(AcquireCC.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.Z, Condition.NZ, Condition.TRUE}
